interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,956 @@
+        species_weights = []
+        for key in species.keys():
+            index = solution.species_index(key)
+            species_weights.append(solution.molecular_weights[index])
+
+        # Append coefficient to species
+        species_list = []
+        for species_text, coef in species.items():
+            if coef == 1.0:
+                species_list.append(species_text)
+            else:
+                species_list.append(f"{coef:.0f} {species_text}")
+
+        species = species_list
+
+        # Reorder species based on molecular weights
+        species = [x for y, x in sorted(zip(species_weights, species))][::-1]
+        reaction_txt.append(" + ".join(species) + third_body)
+
+    reaction_txt = reaction_direction.join(reaction_txt)
+
+    return reaction_txt
+
+
+def eformat(f, precision=7, exp_digits=3):
+    """Formats numbers in a pretty format for printing
+
+    Args:
+        f (float): input float to be formatted into str
+        precision (int, optional): Sets the precision of the str. Defaults to 7.
+        exp_digits (int, optional): Sets the number of digits in the exponent. 
+            Defaults to 3.
+
+    Returns:
+        str: input number formatted for adding to files
+    """    
+    s = f"{f: .{precision}e}"
+    if s == " inf" or s == "-inf":
+        return s
+    else:
+        mantissa, exp = s.split("e")
+        exp_digits += 1  # +1 due to sign
+        return f"{mantissa}E{int(exp):+0{exp_digits}}"
+
+
+def build_arrhenius(reaction_equation, rate, reaction_order, reaction_type):
+    """Builds Arrhenius coefficient string based on reaction type.
+    Parameters
+    ----------
+    rate : ct.Arrhenius
+        Arrhenius-form reaction rate coefficient
+    reaction_order : int or float
+        Order of reaction (sum of reactant stoichiometric coefficients)
+    reaction_type : {ct.ElementaryReaction, ct.ThreeBodyReaction, ct.PlogReaction}
+        Type of reaction
+    Returns
+    -------
+    str
+        String with Arrhenius coefficients
+    """
+    if reaction_type in [ct.ElementaryReaction, ct.PlogReaction]:
+        pre_exponential_factor = rate.pre_exponential_factor * 1e3 ** (
+            reaction_order - 1
+        )
+
+    elif reaction_type == ct.ThreeBodyReaction:
+        pre_exponential_factor = rate.pre_exponential_factor * 1e3**reaction_order
+
+    elif reaction_type in [ct.FalloffReaction, ct.ChemicallyActivatedReaction]:
+        raise ValueError(
+            "Function does not support falloff or chemically activated reactions"
+        )
+    else:
+        raise NotImplementedError(
+            f"Reaction type not supported: ({reaction_type}) {reaction_equation}"
+        )
+
+    activation_energy = rate.activation_energy / CALORIES_CONSTANT
+    arrhenius = [
+        f"{eformat(pre_exponential_factor)}",
+        f"{eformat(rate.temperature_exponent)}",
+        f"{eformat(activation_energy)}",
+    ]
+    return "   ".join(arrhenius)
```

Comment:
How much is this a sticking point for you? These would get pretty ugly if I started to do string templates everywhere because there's just so many coefficients. I did alter these so that they are put together using joins instead of adding to the string where possible.